Tests/UnitTests/FileShareService.DesktopClientTests/Modules/Search/BatchDetailsViewModelTest.cs

```diff
@@ -0,0 +1,90 @@
+﻿using FakeItEasy;
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Abstractions.TestingHelpers;
+using System.Linq;
+using System.Net.Http;
+using System.Reflection;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Windows;
+using UKHO.FileShareClient.Models;
+using UKHO.FileShareService.DesktopClient;
+using UKHO.FileShareService.DesktopClient.Modules.Search;
+using UKHO.FileShareAdminClient;
+
+namespace FileShareService.DesktopClientTests.Modules.Search
+{
+    [TestFixture]
+    public class BatchDetailViewModelTest
+    {
+        private IFileShareApiAdminClientFactory fakeFileShareApiAdminClientFactory = null!;
+        private IMessageBoxService fakeMessageBoxService = null!;
+        private  IFileService fakeFileService = null!;
+        private IFileShareApiAdminClient fakeFileShareApiAdminClient = null!;
+
+        [SetUp]
+        public void Setup()
+        {
+            fakeFileShareApiAdminClientFactory = A.Fake<IFileShareApiAdminClientFactory>();
+            fakeMessageBoxService = A.Fake<IMessageBoxService>();
+            fakeFileService = A.Fake<IFileService>();
+            fakeFileShareApiAdminClient = A.Fake<IFileShareApiAdminClient>();
+        }
+        [Test]
+        public async Task TestDownloadFileWithFileExistsAndYesResponse()
```

Comment:
I'm sorry to create work for you Rashi, but we need the ability to inject SaveFileDialog as a service to allow more of the BatchDetailViewModel to be tested. 
With SaveFileDialog injected as a service we can then test the complete OnDownloadExecutionCommand process

Reply:
Resolved